src/components/smartSearch/EditTargetDialog.tsx

```diff
@@ -0,0 +1,128 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { useState } from 'react';
+import { Box, Button, ButtonBase, Card, CardContent, Dialog, DialogContent, Typography } from '@material-ui/core';
+
+import All from './filters/All';
+import Filter from './Filter';
+import { isFilterWithId } from './utils';
+import MostActive from './filters/MostActive';
+import patchTaskTarget from 'fetching/tasks/patchTaskTarget';
+import { useRouter } from 'next/router';
+import { ZetkinSmartSearchFilter } from 'types/zetkin';
+import { FILTER_TYPE, ZetkinSmartSearchFilterWithId } from 'types/smartSearch';
+import { useMutation, useQueryClient } from 'react-query';
+
+interface EditTargetDialogProps {
+    filterSpec: ZetkinSmartSearchFilter[];
+    onDialogClose: () => void;
+    open: boolean;
+}
+
+const EditTargetDialog = ({ onDialogClose, open, filterSpec }: EditTargetDialogProps) : JSX.Element => {
+    const queryClient = useQueryClient();
+    const { orgId, taskId } = useRouter().query;
+    const filtersWithIds = filterSpec.map((filter, index) => ({ ...filter, id: index }));
+    const [filterArray, setFilterArray] = useState<ZetkinSmartSearchFilterWithId[]>(filtersWithIds);
+    const [selectedType, setSelectedType] = useState<FILTER_TYPE | null>(null);
+    const [selectedFilter, setSelectedFilter] = useState<ZetkinSmartSearchFilterWithId| null>(null);
+
+    const taskMutation = useMutation(patchTaskTarget(orgId as string, taskId as string),{
+        onSettled: () => queryClient.invalidateQueries(['filter_spec', orgId, taskId]),
+    } );
+
+    const handleDialogClose = () => {
+        setSelectedType(null);
+        onDialogClose();
+    };
+
```

Comment:
I think this could be called `closeDialog` and used everywhere `onDialogClose` everywhere else this is used in this component?

Reply:
Can you clarify this grammar?